packages/replay-next/src/suspense/AnalysisCache.ts

```diff
@@ -71,64 +72,79 @@ export function createAnalysisCache<
     comparePoints: compareNumericStrings,
     load: async (begin, end, client, ...paramsWithCacheLoadOptions) => {
       const params = paramsWithCacheLoadOptions.slice(0, -1) as TParams;
-      const points = await findPoints(client, begin, end, ...params);
+      let points: PointDescription[];
+      try {
+        points = await findPoints(client, begin, end, ...params);
+      } catch (error) {
+        if (isCommandError(error, ProtocolError.FocusWindowChange)) {
+          pointsIntervalCache.abort(client, ...params);
+        }
+        throw error;
+      }
       onPointsReceived?.(points);
 
       const evaluationParams = await createEvaluationParams(client, points, ...params);
 
-      client.runEvaluation(
-        {
-          selector: evaluationParams.selector,
-          expression: evaluationParams.expression,
-          frameIndex: evaluationParams.frameIndex,
-          fullPropertyPreview: true,
-          limits: { begin, end },
-        },
-        async results => {
-          for (const result of results) {
-            setPointAndTimeForPauseId(result.pauseId, result.point);
-            cachePauseData(client, result.pauseId, result.data);
+      client
+        .runEvaluation(
```

Comment:
Shouldn't we be _awaiting_ `client.runEvaluation` ?

For one, that method is async (and accepts an async callback) so not _awaiting_ it would cause our interval cache `load` method to signal completion prematurely.

But also, if we aren't awaiting then re-throwing a potential `error` below won't do anything. (It will be a global/uncaught error.)

Reply:
> not awaiting it would cause our interval cache load method to signal completion prematurely.

The interval cache stores the points from `findPoints()`, so we `await` only that, i.e. we *want* this load method to return as soon as points have been received. `runEvaluation()` will fetch the results, which are not stored in the interval cache but in the externally managed results cache.

> if we aren't awaiting then re-throwing a potential error below won't do anything

Well, it should trigger the error overlay in dev and send the error to Sentry in prod.

Reply:
>> if we aren't awaiting then re-throwing a potential error below won't do anything
>
> Well, it should trigger the error overlay in dev and send the error to Sentry in prod.

I think this kind of only prevents the worst case though (no logging at all). Code calling this cache would expect it to fail in a way that would trigger an error boundary (or `try/catch`).

Interval caches are expected to block. (This isn't a _streaming cache_.)

Reply:
Ah, but that also means that the request is already resolved in the `pointsIntervalCache` when `runEvaluation` throws, so `abort` won't work and I need to use `evict` instead.

Reply:
> Code calling this cache would expect it to fail in a way that would trigger an error boundary

I would not expect `pointsIntervalCache.read()` to throw if the points were fetched successfully. But `resultsCache.read()` should throw in this case, so we should do something like `points.forEach(point => resultsCache.cacheError(error, point, ...params);` in the catch block. The suspense package doesn't offer a method for caching errors yet though.

Reply:
Ah, that makes sense. Give me a few minutes to think on that.

Reply:
https://github.com/replayio/devtools/pull/9086